src/components/app/data/hooks/useCourseRedemptionEligibility.js

```diff
@@ -1,34 +1,48 @@
 import { useParams } from 'react-router-dom';
 import { useQuery } from '@tanstack/react-query';
 
+import { isRunUnrestricted } from '../utils';
 import useCourseMetadata from './useCourseMetadata';
 import { queryCanRedeem } from '../queries';
 import useEnterpriseCustomer from './useEnterpriseCustomer';
 import useLateEnrollmentBufferDays from './useLateEnrollmentBufferDays';
+import useEnterpriseCustomerContainsContent from './useEnterpriseCustomerContainsContent';
 
 export function transformCourseRedemptionEligibility({
   courseMetadata,
   canRedeemData,
   courseRunKey,
+  restrictedRunsAllowed,
 }) {
-  const redeemabilityForActiveCourseRun = canRedeemData.find(r => r.contentKey === courseMetadata.activeCourseRun?.key);
+  // Begin by excluding restricted runs that should not be visible to the requester.
+  // This filtering does not control visibility of individual course runs, but
+  // it does serve as input to the determination of redemption eligiblity.
+  const unrestrictedCanRedeemData = canRedeemData.filter(r => isRunUnrestricted({
```

Comment:
FWIW, I did intentionally decide to pass the restricted runs to `can-redeem` in an effort to minimize code complexity, but with the understanding that it would make those specific course about pages load more slowly.  That said, I'm not married to the simpler code, and performance may be a requirement here.

Reply:
There's a performance tradeoff with both options:

1. Calling `can-redeem` with all (available) restricted runs forces processing of runs that might not end up visible (unnecessary work for `can-redeem`). No request waterfall introduced, beyond the existing waterfall with `/api/v1/courses/` and `can-redeem` API calls. A course with a lot of restricted runs that aren't configured for a customer's catalogs would be processed (ideally it's just 1-2 addtl runs, but is this guaranteed?). It's a no-op for courses without restricted runs.
2. Calling `can-redeem` with only the allowed restricted runs for more performant `can-redeem`, but with the request waterfall where we need to resolve `contains_content_items` first to discern allowed restricted runs before calling `can-redeem`. Introduced query dependency has slight performance implications for courses without restricted runs due to always fulfilling the request waterfall of `contains_content_items` -> `can-redeem`.

While I don't feel too strongly one way or another, I think I might lean towards (1) upon further reflection this morning (i.e., no filtering of restricted runs before the `can-redeem` API request in the `courseLoader`) since it's a no-op for courses without restricted runs (majority / happy path case), _with the assumption that courses with restricted runs only have 1-2 addtl runs, not a large N._

Reply:
Exactly, the common case is that a course has no restricted runs, and when they do have restricted runs I imagine there would only ever be 1-2 available (e.g. we could run 100s of restricted runs over many years, but only a couple actually "available"). This is worth validating, but seems the most probable.

Reply:
Confirmed with team during standup: a course isunlikely to contain more than a couple restricted runs at any given time.